### PR TITLE
fix for 21476

### LINKED ISSUE #21476: When essential claim check fails the error message should provide detailed information
### Description

When we configure an external IDP with essential claim settings (see #20191) and the authentication process fails because of this check, the login error displays a generic message `Unexpected error when authenticating with identity provider`.

The message should instead be more detailed and provide troubleshooting information, like: 
`The ID token issued by the identity provider {1} did not match the configured essential claim. Please contact your administrator`

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by dmartinol**
Closes #21476 
Added specific exception to throw a new error message identified for this issue

### CODE REVIEW

File: server-spi-private/src/main/java/org/keycloak/broker/provider/IdentityBrokerUnmatchedEssentialClaimException.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.broker.provider;
+
+/**
+ * @author dmartino
+ */
+public class IdentityBrokerUnmatchedEssentialClaimException extends RuntimeException {
```
Comment (pedroigor) [inline server-spi-private/src/main/java/org/keycloak/broker/provider/IdentityBrokerUnmatchedEssentialClaimException.java:22]: Do we really need this exception class? Can we just leverage `IdentityBrokerException` and handle it accordingly to build the error response?

File: services/src/main/java/org/keycloak/broker/oidc/OIDCIdentityProvider.java [hunk 2]
```diff
@@ -432,6 +433,8 @@ public BrokeredIdentityContext getFederatedIdentity(String response) {
             }
 
             return identity;
+        } catch (IdentityBrokerUnmatchedEssentialClaimException e) {
```
Comment (pedroigor) [inline services/src/main/java/org/keycloak/broker/oidc/OIDCIdentityProvider.java:436]: You would just re-throw `IdentityBrokerException` as per my first comment.

Comment (dmartinol) [reply services/src/main/java/org/keycloak/broker/oidc/OIDCIdentityProvider.java]: Trying to carry the error message in the existing exception in the next commit

**Review by pedroigor — COMMENTED** [review]
@dmartinol Thanks, I think the remaining task is test coverage. I think you are already familiar with `KcOidcBrokerTest`, right?

**Review by pedroigor — APPROVED** [review]
@dmartinol Thanks!

**Pull request merged**